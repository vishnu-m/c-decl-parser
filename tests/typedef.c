typedef void DRAWF( int, int );

typedef struct club
{
        char name[30];
        int size, year;
} GROUP;

typedef GROUP *PG;

typedef char char_t, *char_p, (*fp)(void);