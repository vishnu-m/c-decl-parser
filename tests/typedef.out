TypedefDecl		underlying type = "void (int, int)"		type = "DRAWF"
StructDecl	"club"		Parent = "typedef.c"
FieldDecl	"name"		type = "char [30]"		Parent = "club"
FieldDecl	"size"		type = "int"		Parent = "club"
FieldDecl	"year"		type = "int"		Parent = "club"
TypedefDecl		underlying type = "struct club"		type = "GROUP"
TypedefDecl		underlying type = "GROUP *"		type = "PG"
TypedefDecl		underlying type = "char"		type = "char_t"
TypedefDecl		underlying type = "char *"		type = "char_p"
TypedefDecl		underlying type = "char (*)(void)"		type = "fp"
